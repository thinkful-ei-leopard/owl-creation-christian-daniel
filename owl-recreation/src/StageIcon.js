import React from 'react';

function StageIcon(participant) {
    return (
        <div className='stageIcon'>
            <img alt='stage icon' src={participant.avatar} />
            <p>{participant.name}</p>
            <div className={participant.onStage ? 'onStage' : 'notOnStage'}></div>
        </div>
    )
}

export default StageIcon;