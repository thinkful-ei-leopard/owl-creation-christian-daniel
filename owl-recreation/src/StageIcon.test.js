import React from 'react';
import ReactDOM from 'react-dom';
import StageIcon from './StageIcon';
import renderer from 'react-test-renderer';

describe('<StageIcon />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<StageIcon />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<StageIcon id='2' name='Caty Flucker' avatar='https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1' onStage='false' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});