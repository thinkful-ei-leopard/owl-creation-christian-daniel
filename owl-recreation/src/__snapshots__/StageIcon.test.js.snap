// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageIcon /> renders the UI as expected 1`] = `
<div
  className="stageIcon"
>
  <img
    alt="stage icon"
    src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
  />
  <p>
    Caty Flucker
  </p>
  <div
    className="onStage"
  />
</div>
`;
